pkg/diagnostics/systemd/locate_units.go

```diff
@@ -3,15 +3,35 @@ package systemd
 import (
 	"fmt"
 	"os/exec"
+	"runtime"
 	"strings"
 
 	"github.com/openshift/origin/pkg/diagnostics/log"
 	"github.com/openshift/origin/pkg/diagnostics/types"
 )
 
+// IsSystemd checks if this is a systemd host. We do this by checking that both systemctl and journald are
+// present and functional. When running in a container it is possible for systemctl to be
+// present but non-functional, or misleadingly functional due to sys mounts and a privileged container.
+func IsSystemd() bool {
+	if runtime.GOOS == "linux" {
+		systemctlErr := exec.Command("systemctl").Run()
+		journalctlErr := exec.Command("journalctl", "-n", "1").Run()
```

Comment:
Arguably, we're not as interested in "is this a systemd host" as in "Can I read the host systemd unit status with systemctl" and "Can I read the journald logs with journalctl". Are those always going to have the same answer? Even for the simple host case, not necessarily - a non-root user can get systemd unit status, but can't read journald logs. Not sure if the same may be true in a container. I could see admitting a little more nuance here, since you're already refining the conditions a bit.